source/mqtt_agent.c

```diff
@@ -699,7 +704,13 @@ static MQTTStatus_t createAndAddCommand( CommandType_t commandType,
             {
                 /* Could not send the command to the queue so release the command
                  * structure again. */
-                pMqttAgentContext->agentInterface.releaseCommand( pCommand );
+                commandReleased = pMqttAgentContext->agentInterface.releaseCommand( pCommand );
```

Comment:
We assert in mqtt_agent_command_functions.c:239 if `releaseCommand` isn't defined, should this function do the same? e.g.:

```c
assert( pMqttAgentContext->agentInterface.releaseCommand != NULL );
```

Reply:
No need, as long as you've validated everything beforehand.